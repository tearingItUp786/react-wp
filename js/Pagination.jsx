import React from 'react';
import { Link } from 'react-router-dom';

type Props = {
  searchTerm: string,
  categories: string,
  authors: string,
  totalPages: number,
  currentPage: number
};

const Pagination = (props: Props) => {
  const pageLinkElementsArray = [];

  const { searchTerm, categories, authors, totalPages, currentPage } = props;
  let linkTo = `/search?searchTerm=${searchTerm}`;

  if (categories !== undefined) {
    linkTo = `${linkTo}&categories=${categories}`;
  }

  if (authors !== undefined) {
    linkTo = `${linkTo}&author=${authors}`;
  }

  for (let i = 1; i <= totalPages; i += 1) {
    let listItemClass = 'page-item';
    if (i === currentPage) {
      listItemClass = `${listItemClass} active`;
    }
    pageLinkElementsArray.push(
      <li key={i} className={listItemClass}>
        <Link className="page-link" to={`${linkTo}&page=${i}`}>
          {i}
        </Link>
      </li>
    );
  }

  return (
    <nav aria-label="pagination">
      <ul className="pagination">{pageLinkElementsArray}</ul>
    </nav>
  );
};

export default Pagination;
