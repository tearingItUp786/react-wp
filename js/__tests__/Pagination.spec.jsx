import React from 'react';
import { BrowserRouter, Link } from 'react-router-dom';
import { configure, shallow, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Pagination from '../Pagination';

configure({ adapter: new Adapter() });

const props = {
  searchTerm: '',
  categories: '',
  authors: '',
  totalPages: 5,
  currentPage: 2
};

describe('Testing pagination component', () => {
  test('Pagination renders properly', () => {
    const component = shallow(<Pagination {...props} />);
    expect(component).toMatchSnapshot();
  });

  test('Pagination should render correct amount of Link Tags', () => {
    const component = shallow(<Pagination {...props} />);
    expect(component.find(Link).length).toEqual(props.totalPages);
  });

  test('List item with active class should have inner text that matches current page prop', () => {
    const component = render(
      <BrowserRouter>
        <Pagination {...props} />
      </BrowserRouter>
    );

    const { currentPage } = props;

    expect(
      component
        .find('.active')
        .children()
        .text()
    ).toEqual(currentPage.toString());
  });
});
