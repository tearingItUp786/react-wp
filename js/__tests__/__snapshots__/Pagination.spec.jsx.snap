// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing pagination component Pagination renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pagination
    authors=""
    categories=""
    currentPage={2}
    searchTerm=""
    totalPages={5}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "aria-label": "pagination",
      "children": <ul
        className="pagination"
>
        <li
                className="page-item"
        >
                <Link
                        className="page-link"
                        replace={false}
                        to="/search?searchTerm=&categories=&author=&page=1"
                >
                        1
                </Link>
        </li>
        <li
                className="page-item active"
        >
                <Link
                        className="page-link"
                        replace={false}
                        to="/search?searchTerm=&categories=&author=&page=2"
                >
                        2
                </Link>
        </li>
        <li
                className="page-item"
        >
                <Link
                        className="page-link"
                        replace={false}
                        to="/search?searchTerm=&categories=&author=&page=3"
                >
                        3
                </Link>
        </li>
        <li
                className="page-item"
        >
                <Link
                        className="page-link"
                        replace={false}
                        to="/search?searchTerm=&categories=&author=&page=4"
                >
                        4
                </Link>
        </li>
        <li
                className="page-item"
        >
                <Link
                        className="page-link"
                        replace={false}
                        to="/search?searchTerm=&categories=&author=&page=5"
                >
                        5
                </Link>
        </li>
</ul>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="page-item"
>
            <Link
                        className="page-link"
                        replace={false}
                        to="/search?searchTerm=&categories=&author=&page=1"
            >
                        1
            </Link>
</li>,
          <li
            className="page-item active"
>
            <Link
                        className="page-link"
                        replace={false}
                        to="/search?searchTerm=&categories=&author=&page=2"
            >
                        2
            </Link>
</li>,
          <li
            className="page-item"
>
            <Link
                        className="page-link"
                        replace={false}
                        to="/search?searchTerm=&categories=&author=&page=3"
            >
                        3
            </Link>
</li>,
          <li
            className="page-item"
>
            <Link
                        className="page-link"
                        replace={false}
                        to="/search?searchTerm=&categories=&author=&page=4"
            >
                        4
            </Link>
</li>,
          <li
            className="page-item"
>
            <Link
                        className="page-link"
                        replace={false}
                        to="/search?searchTerm=&categories=&author=&page=5"
            >
                        5
            </Link>
</li>,
        ],
        "className": "pagination",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="page-link"
              replace={false}
              to="/search?searchTerm=&categories=&author=&page=1"
>
              1
</Link>,
            "className": "page-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": 1,
              "className": "page-link",
              "replace": false,
              "to": "/search?searchTerm=&categories=&author=&page=1",
            },
            "ref": null,
            "rendered": 1,
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="page-link"
              replace={false}
              to="/search?searchTerm=&categories=&author=&page=2"
>
              2
</Link>,
            "className": "page-item active",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": 2,
              "className": "page-link",
              "replace": false,
              "to": "/search?searchTerm=&categories=&author=&page=2",
            },
            "ref": null,
            "rendered": 2,
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="page-link"
              replace={false}
              to="/search?searchTerm=&categories=&author=&page=3"
>
              3
</Link>,
            "className": "page-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": 3,
              "className": "page-link",
              "replace": false,
              "to": "/search?searchTerm=&categories=&author=&page=3",
            },
            "ref": null,
            "rendered": 3,
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="page-link"
              replace={false}
              to="/search?searchTerm=&categories=&author=&page=4"
>
              4
</Link>,
            "className": "page-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": 4,
              "className": "page-link",
              "replace": false,
              "to": "/search?searchTerm=&categories=&author=&page=4",
            },
            "ref": null,
            "rendered": 4,
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="page-link"
              replace={false}
              to="/search?searchTerm=&categories=&author=&page=5"
>
              5
</Link>,
            "className": "page-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": 5,
              "className": "page-link",
              "replace": false,
              "to": "/search?searchTerm=&categories=&author=&page=5",
            },
            "ref": null,
            "rendered": 5,
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-label": "pagination",
        "children": <ul
          className="pagination"
>
          <li
                    className="page-item"
          >
                    <Link
                              className="page-link"
                              replace={false}
                              to="/search?searchTerm=&categories=&author=&page=1"
                    >
                              1
                    </Link>
          </li>
          <li
                    className="page-item active"
          >
                    <Link
                              className="page-link"
                              replace={false}
                              to="/search?searchTerm=&categories=&author=&page=2"
                    >
                              2
                    </Link>
          </li>
          <li
                    className="page-item"
          >
                    <Link
                              className="page-link"
                              replace={false}
                              to="/search?searchTerm=&categories=&author=&page=3"
                    >
                              3
                    </Link>
          </li>
          <li
                    className="page-item"
          >
                    <Link
                              className="page-link"
                              replace={false}
                              to="/search?searchTerm=&categories=&author=&page=4"
                    >
                              4
                    </Link>
          </li>
          <li
                    className="page-item"
          >
                    <Link
                              className="page-link"
                              replace={false}
                              to="/search?searchTerm=&categories=&author=&page=5"
                    >
                              5
                    </Link>
          </li>
</ul>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="page-item"
>
              <Link
                            className="page-link"
                            replace={false}
                            to="/search?searchTerm=&categories=&author=&page=1"
              >
                            1
              </Link>
</li>,
            <li
              className="page-item active"
>
              <Link
                            className="page-link"
                            replace={false}
                            to="/search?searchTerm=&categories=&author=&page=2"
              >
                            2
              </Link>
</li>,
            <li
              className="page-item"
>
              <Link
                            className="page-link"
                            replace={false}
                            to="/search?searchTerm=&categories=&author=&page=3"
              >
                            3
              </Link>
</li>,
            <li
              className="page-item"
>
              <Link
                            className="page-link"
                            replace={false}
                            to="/search?searchTerm=&categories=&author=&page=4"
              >
                            4
              </Link>
</li>,
            <li
              className="page-item"
>
              <Link
                            className="page-link"
                            replace={false}
                            to="/search?searchTerm=&categories=&author=&page=5"
              >
                            5
              </Link>
</li>,
          ],
          "className": "pagination",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="page-link"
                replace={false}
                to="/search?searchTerm=&categories=&author=&page=1"
>
                1
</Link>,
              "className": "page-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": 1,
                "className": "page-link",
                "replace": false,
                "to": "/search?searchTerm=&categories=&author=&page=1",
              },
              "ref": null,
              "rendered": 1,
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="page-link"
                replace={false}
                to="/search?searchTerm=&categories=&author=&page=2"
>
                2
</Link>,
              "className": "page-item active",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": 2,
                "className": "page-link",
                "replace": false,
                "to": "/search?searchTerm=&categories=&author=&page=2",
              },
              "ref": null,
              "rendered": 2,
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="page-link"
                replace={false}
                to="/search?searchTerm=&categories=&author=&page=3"
>
                3
</Link>,
              "className": "page-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": 3,
                "className": "page-link",
                "replace": false,
                "to": "/search?searchTerm=&categories=&author=&page=3",
              },
              "ref": null,
              "rendered": 3,
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="page-link"
                replace={false}
                to="/search?searchTerm=&categories=&author=&page=4"
>
                4
</Link>,
              "className": "page-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": 4,
                "className": "page-link",
                "replace": false,
                "to": "/search?searchTerm=&categories=&author=&page=4",
              },
              "ref": null,
              "rendered": 4,
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "5",
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="page-link"
                replace={false}
                to="/search?searchTerm=&categories=&author=&page=5"
>
                5
</Link>,
              "className": "page-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": 5,
                "className": "page-link",
                "replace": false,
                "to": "/search?searchTerm=&categories=&author=&page=5",
              },
              "ref": null,
              "rendered": 5,
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
